citesphere/src/test/java/edu/asu/diging/citesphere/core/service/impl/AuthorityServiceTest.java

```diff
@@ -238,11 +309,281 @@ public void test_save() {
+            Assert.assertTrue(entry.getName().contains("Einstein"));
+        }
+
+    }
+
+    @Test
+    public void test_findByName_withFullName() {
+
+        // Testing authority search with full name 'Einstein Albert'
+        entriesAlbert.remove(1);
+        Mockito.when(authorityRepository.findByUsernameAndNameContainingAndNameContainingOrderByName(user.getUsername(),
+                "Einstein", "Albert", paging)).thenReturn(entriesAlbert);
+
+        List<IAuthorityEntry> searchFullName = managerToTest.findByName(user, "Einstein", "Albert", page, pageSize);
+
+        for (IAuthorityEntry entry : searchFullName) {
+            Assert.assertTrue(entry.getName().contains("Einstein") && entry.getName().contains("Albert"));
+        }
+
+        // Testing authority search by full name 'Jane Rose'
+        Mockito.when(authorityRepository.findByUsernameAndNameContainingAndNameContainingOrderByName(user.getUsername(),
+                "Rose", "Jane", paging)).thenReturn(entriesJane);
+
+        List<IAuthorityEntry> searchFullNameJane = managerToTest.findByName(user, "Rose", "Jane", page, pageSize);
+
+        Assert.assertFalse(searchFullNameJane.isEmpty());
+        Assert.assertTrue(searchFullNameJane.get(0) != null && searchFullNameJane.size() == 1);
+        Assert.assertTrue(searchFullNameJane.get(0).getName().contains("Jane"));
+        Assert.assertFalse(searchFullNameJane.get(0).getName().contains("Albert"));
+
+    }
+
+    @Test
+    public void test_findByName_withInterchangedFirstNameAndLastName() {
+
+        // Testing authority search by full name with first and last name interchanged
+        List<IAuthorityEntry> searchFullNameReverseOrder = managerToTest.findByName(user, "Albert", "Einstein", page,
+                pageSize);
+
+        for (IAuthorityEntry entry : searchFullNameReverseOrder) {
+            Assert.assertTrue(entry.getName().contains("Einstein") && entry.getName().contains("Albert"));
+        }
+
+    }
+
+    @Test
+    public void test_findByName_resultsNotFound() {
+
+        Mockito.when(authorityRepository.findByUsernameAndNameContainingAndNameContainingOrderByName(user.getUsername(),
+                "Peter", "Doe", paging)).thenReturn(new ArrayList<>());
+
+        // Testing authority search by full name, name not found
+        List<IAuthorityEntry> actual3 = managerToTest.findByName(user, "Peter", "Doe", page, pageSize);
+
+        Assert.assertTrue(actual3.isEmpty());
+    }
+
+    @Test
+    public void test_getTotalUserAuthoritiesPages() {
+
+        // testing total number of user authority pages when search authority has last
+        // name as "Albert" and pageSize is 10
+        Mockito.when(authorityRepository
+                .countByUsernameAndNameContainingAndNameContainingOrderByName(user.getUsername(), "", "Albert"))
+                .thenReturn((long) 3);
+        Assert.assertTrue(managerToTest.getTotalUserAuthoritiesPages(user, "", "Albert", pageSize) == 1);
```

Comment:
use assertEquals for all these, so that the test cases would say something like : Expected: 1, actual: 2 if they fail